### ISSUE TITLE
[BUG] Changing syncdex status to "Dropped" doesn't work on first attempt.

### ISSUE THREAD

**Issue by Jerrk** [Glagan/SyncDex#36]
**Describe the bug**
this bug always happens after changing the status AFTER reading the latest chapter.

If just navigating to the title page in another way it only happens sometimes.

**To Reproduce**
1. Go to any manga title page
2. add to reading
3. refresh the page and make sure it's tracked within syncdex
4. read the latest chapter and get redirected to the title page
5. syncdex -> edit -> status: dropped -> save
6. repeat step 5 again for the status to actually change

**Expected behavior (optional)**
i should only have to do step 5 once

**Screenshots**
here is a really bad example where i have to change the status multiple time for it to stick, sometimes it disappears from the syncdex status completely as if i've removed it completely instead of changing status to dropped.
https://i.imgur.com/dwjumAK.gif

**Browser**
Firefox 86.0a1 (2021-01-23) (64-bit)

**Options and logs**
[bighonkinglog.txt](https://github.com/Glagan/SyncDex/files/5860925/bighonkinglog.txt)

**Additional context**
with this big of a debug log the settings page becomes completely unresponsive while trying to copy it debug output. is the dropbox logging really necessary? haha

**Comment by Glagan** [comment]
Sorry about the logs, I forgot to prune them if they were too big, I'll add that.  
There's also some debug logs that shouldn't be here, I'll fix that.

For the actual issue, I tried it (only once) but it's working for me, I think it's because you edit too early and not all initial requests are received.

If you disable the overview only for your main service you'll see them all loading.

![image](https://user-images.githubusercontent.com/8447621/105613720-afd6df80-5dc4-11eb-824f-7d8fa8a2b08a.png)

But that's not expected, the SyncDex overview should still be blocked until everything is received to prevent that from happening, I'll add (or fix ?) that.

**Comment by Glagan** [comment]
It's actually **MangaUpdates** being broken if you go from one list to another in some cases, on top of the other issue.

I'm doing more tests and I'll probably publish the update later or tomorrow.

**Comment by Glagan** [comment]
Should be fixed in ``0.2.4``, I also recommend you to remove some Services for a better experience, and I can't recommend to use **AnimePlanet** and **MangaUpdates**, I added them but I now wonder why, they don't really have an API, they don't track as much as the others (most importantly dates, and other Informations that for now can't be updated within SyncDex) and are (sometimes) slower.

Also, I'll copy/paste what I wrote in the release, but essentially **MangaUpdates** is making everything slower (in some cases):

>Their **API** (at least what is exposed and used on their site) is not mean to be used outside of the site itself and is missing some informations in some cases, for example if a Title is in any other list that **Reading**, it will always trigger an update since the progress is **not** displayed.

**Comment by Jerrk** [comment]
It appears with the latest update the tracking will not update for the latest read chapter.
https://i.imgur.com/DFoMuCU.mp4

I took your advice and stopped using animeplanet and mangaupdates to speed up the processing, i haven't really been using MU for years and i just started using animeplanet with syncdex

**Comment by Glagan** [comment]
I think you're just going too fast, if you display the bar on the right you'll see icons for each active services:

![image](https://user-images.githubusercontent.com/8447621/105637800-93dc4800-5e6f-11eb-8cf1-fed1848a18e3.png)

When you're reading, I have to fetch each services at least once, compare and assign the new values, then send another request to each Services.  
Each of these requests can take up to a few seconds and I can't do anything about that sadly.

You need to wait until you see a summary notification or at least a notification telling you that progress was not *not* updated if you have some options enabled (Confirm chapter and any Save only if ...).

**Comment by Jerrk** [comment]
for this specific video i took i went through it fast.

but when actually reading the chapter it can take a minute or two and it STILL doesn't update... so i doubt me just being fast is the problem.

heres another video where you can see  the green outline and the status update message
https://i.imgur.com/qVTychR.mp4

i did not have this issue in 0.2.3

**Comment by Jerrk** [comment]
If i enable the option "Confirm Chapter" it will tell me that the title is not in my mangadex list even though it definitely is, so there appears to have been an issue somewhere in the latest version?

edit: if i disable "Update only if in list" i dont get this issue either, so somehow it's not seeing the manga as being in my mangadex list somehow

**Comment by Glagan** [comment]
Ho, I wasn't looking at that but you do have the ``updateOnlyInList`` option enabled.  
It was actually bugged and I just published an update (``0.2.5``) because I was using the wrong API url to check the status.

Try again when ``0.2.5`` is out (should already be, if you manually check for updates) and tell me if it's fixed.

**Comment by Jerrk** [comment]
Yes everything appears to be in order now 🙂 

I noticed another issue when i removed MU and AP that you might wanna take a look at.

if you somehow click the activate button multiple times due to lag or double-click or what-have-you, it will add the service multiple times within the reader status check. 
![firefox_QMXLQyLwgE](https://user-images.githubusercontent.com/49324581/105638880-85912a80-5e75-11eb-9c62-6197f38cf6e3.png)
(i over-exaggerated it with like 10 clicks to fill up the screen, but you get the idea)
Visually i cannot find other places where this causes a problem but i'm sure it is not intended and should probably be fixed

**Comment by Glagan** [comment]
No, that's not intended ahah, I tried it and the Activate button just instantly go away for me, but I'll add a check to avoid that.
